### PR TITLE
Filter case prop

### PR DESCRIPTION
**Pull request by andrey-semin**
Added a property to enable case sensitive filtering (#5)

### CODE REVIEW

File: dist/js/sth-select.js [hunk 1]
```diff
@@ -195,15 +195,16 @@
...
 			_clear();
 
 			var rerenderOnEachItem = false;
 			var $listItems = $([]);
-			var textFilter = _$filter.val().toLowerCase();
+			var textFilter = formatText(caseSensitive, _$filter.val());
 
-			_items.map(function (item) {
-				if (item.text.toLowerCase().indexOf(textFilter) != -1) {
+			_items.forEach(function (item) {
```
Comment (andrey-semin) [inline dist/js/sth-select.js]: We should use `forEach` here instead of map because `map` returns new array and we dont need it here

Comment (StanleySathler) [reply dist/js/sth-select.js]: Great! You're correct. 👍

**Comment by StanleySathler** [comment]
Hey @andrey-semin 
Awesome! We loved your contribution! 🎉 

First, I'd like to ask your opinion about renaming the method `formatText()` to `_formatText()`, keeping the same pattern. What you think about it?

And second, what about checking the `_properties.caseSensitive` directly inside of `_renderList()` instead of passing it as a parameter?

We'd love to hear your opinion about both of these points. 💪 💪

**Comment by andrey-semin** [comment]
> First, I'd like to ask your opinion about renaming the method formatText() to _formatText(), keeping the same pattern. What you think about it?

If it's a pattern across the project then we should keep it consistent =, will update tomorrow

> And second, what about checking the _properties.caseSensitive directly inside of _renderList() instead of passing it as a parameter?

I prefer to pass data to functions rather then working with some outside state. It gives more control, testability and is a kind-of-best-practice (see `pure functions` and `functional programming`

**Comment by StanleySathler** [comment]
> I prefer to pass data to functions rather then working with some outside state. It gives more control, testability and is a kind-of-best-practice (see pure functions and functional programming

Great! That's why I love the open source community: everyday we have the chance to learn something new. 🎉 So, in this case, I'll probably be applying some changes based on these arguments to the rest of the functions.

Since you said you agree about renaming `formatText()` to `_formatText()`, I'll be waiting the rename to accept the pull request.

**Comment by andrey-semin** [comment]
updated

**Comment by StanleySathler** [comment]
Awesome, @andrey-semin! Thank you for your Pull Request and thank you for your enriching comments. Yay! 🎉

**Pull request merged**